pkg/volume/plugins.go

```diff
@@ -55,6 +55,10 @@ type VolumeOptions struct {
 	ClusterName string
 	// Tags to attach to the real volume in the cloud provider - e.g. AWS EBS
 	CloudTags *map[string]string
+	// ProvisioningOptions are additional options for the provisioner, e.g.
+	// to create a volume of certain QoS characteristics. These options are
+	// opaque to anything but appropriate provisioner plugin.
+	ProvisioningOptions string
```

Comment:
By "expose internals", I mean to code.

The admin, as a person, needs to understand the params.  That is gleaned through docs.

The system, on the other hand, does not know what those params are and does not care. They are opaque to the system.

Passing a map instead of a string allows the system to set keys and values, which is the exposure I was referring to.  I don't know if this is semantically right or wrong (nor do I have a strong preference).